JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -10,91 +10,309 @@
-
-    @Option(names = { "-s", "--client-secret" }, description = "Client Secret")
-    private String clientSecret = "";
-
-    private static Connection connectToDB(String host, String port, String dbName, String accessToken,
-            String refreshToken, String clientSecret) throws SQLException {
-        Properties jdbcOptions = new Properties();
-        jdbcOptions.put("oauthaccesstoken", accessToken);
-        jdbcOptions.put("oauthrefreshtoken", refreshToken);
-        jdbcOptions.put("oauthclientsecret", clientSecret);
-
-        return DriverManager.getConnection(
-                "jdbc:vertica://" + host + ":" + port + "/" + dbName, jdbcOptions);
-    }
-
-    private static ResultSet executeQuery(Connection conn) throws SQLException {
-        Statement stmt = conn.createStatement();
-        return stmt.executeQuery("SELECT user_id, user_name FROM USERS ORDER BY user_id");
-    }
-
-    private static void printResults(ResultSet rs) throws SQLException {
-        int rowIdx = 1;
-        while (rs.next()) {
-            System.out.println(rowIdx + ". " + rs.getString(1).trim() + " " + rs.getString(2).trim());
-            rowIdx++;
-        }
-    }
-
-    @Override
-    public Integer call() throws Exception {
-        try {
-            Connection conn = connectToDB(host, port, dbName, accessToken, refreshToken, clientSecret);
-            ResultSet rs = executeQuery(conn);
-            printResults(rs);
-            conn.close();
-        } catch (SQLTransientConnectionException connException) {
-            System.out.print("Network connection issue: ");
-            System.out.print(connException.getMessage());
-            System.out.println(" Try again later!");
-        } catch (SQLInvalidAuthorizationSpecException authException) {
-            System.out.print("Could not log into database: ");
-            System.out.print(authException.getMessage());
-            System.out.println(" Check the login credentials and try again.");
-        } catch (SQLException e) {
-            e.printStackTrace();
-        }
-        return 0;
-    }
-
-    public static void main(String[] args) {
-        int exitCode = new CommandLine(new OAuthSampleApp()).execute(args);
-        System.exit(exitCode);
-    }
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.vertica.jdbc.VerticaConnection;
+
+public class OAuthSampleApp {
+	private static Properties prop;
+	private static Map<String, String> csProp = new HashMap<String, String>();
+	private static String OAUTH_ACCESS_TOKEN_VAR_STRING = "OAUTH_SAMPLE_ACCESS_TOKEN";
```

Comment:
In java we declare constants as `final` so the constant properties here should be `private static final`

Constants are also generally grouped together and declared first in a class, so I would move these two constants to be the first properties defined in the class followed by a newline and then followed by the rest of the properties

Reply:
Changed OAUTH_ACCESS_TOKEN_VAR_STRING, OAUTH_REFRESH_TOKEN_VAR_STRING to constant "private static final"